main/src/ca/uwaterloo/flix/language/phase/jvm/GenFunctionInterfaces.scala

```diff
@@ -52,23 +54,35 @@ object GenFunctionInterfaces {
       return None
     }
 
-    // The function arity is simply the number of type arguments.
-    val arity = args.length
+    // `JvmType` of the continuation interface for `tpe`
+    val continuationType = JvmOps.getContinuationType(tpe)
 
-    // The name of the functional interface is of the form:
-    // Fn1$Int$Bool,
-    // Fn2$Int$Int$Bool,
-    // Fn3$Obj$Obj$Obj$Bool, etc.
-    // TODO: Probably use a helper function to compute this?
-    val name = "Fn" + arity + "$"
+    // `JvmType` of the functional interface for `tpe`
+    val functionType = JvmOps.getFunctionType(tpe)
 
-    for (arg <- args) {
-      // do something with the arguments ...
+    // Class visitor
+    val visitor = new ClassWriter(0)
```

Comment:
This classwriter is different. Could it be the same as before? Why/why not?

Reply:
We can use the same as before. In this case there is no difference so I didn't override that method but overriding that method is harmless.